ext/config.m4

```diff
@@ -7,28 +7,92 @@ if test -z "$PHP_GMP"; then
-    types/varint.c types/set.c types/map.c types/collection.c, $ext_shared)
+  CASSANDRA_CLASSES="\
+    src/Cassandra.c \
+    src/Cassandra/BatchStatement.c \
+    src/Cassandra/Cluster.c \
+    src/Cassandra/DefaultCluster.c \
+    src/Cassandra/DefaultSession.c \
+    src/Cassandra/Exception.c \
+    src/Cassandra/ExecutionOptions.c \
+    src/Cassandra/Future.c \
+    src/Cassandra/FutureClose.c \
+    src/Cassandra/FuturePreparedStatement.c \
+    src/Cassandra/FutureRows.c \
+    src/Cassandra/FutureSession.c \
+    src/Cassandra/FutureValue.c \
+    src/Cassandra/PreparedStatement.c \
+    src/Cassandra/Rows.c \
+    src/Cassandra/Session.c \
+    src/Cassandra/SimpleStatement.c \
+    src/Cassandra/SSLOptions.c \
+    src/Cassandra/Statement.c \
+    src/Cassandra/Cluster/Builder.c \
+    src/Cassandra/Exception/AlreadyExistsException.c \
+    src/Cassandra/Exception/AuthenticationException.c \
+    src/Cassandra/Exception/ConfigurationException.c \
+    src/Cassandra/Exception/DomainException.c \
+    src/Cassandra/Exception/ExecutionException.c \
+    src/Cassandra/Exception/InvalidArgumentException.c \
+    src/Cassandra/Exception/InvalidQueryException.c \
+    src/Cassandra/Exception/InvalidSyntaxException.c \
+    src/Cassandra/Exception/IsBootstrappingException.c \
+    src/Cassandra/Exception/LogicException.c \
+    src/Cassandra/Exception/OverloadedException.c \
+    src/Cassandra/Exception/ProtocolException.c \
+    src/Cassandra/Exception/ReadTimeout.c \
+    src/Cassandra/Exception/RuntimeException.c \
+    src/Cassandra/Exception/ServerException.c \
+    src/Cassandra/Exception/TimeoutException.c \
+    src/Cassandra/Exception/TruncateException.c \
+    src/Cassandra/Exception/UnauthorizedException.c \
+    src/Cassandra/Exception/UnavailableException.c \
+    src/Cassandra/Exception/UnpreparedException.c \
+    src/Cassandra/Exception/ValidationException.c \
+    src/Cassandra/Exception/WriteTimeoutException.c \
+    src/Cassandra/SSLOptions/Builder.c \
+  ";
+
+  CASSANDRA_TYPES="\
+    util/bytes.c \
+    util/collections.c \
+    util/consistency.c \
+    util/future.c \
+    util/inet.c \
+    util/math.c \
+    util/ref.c \
+    util/result.c \
+    util/uuid_gen.c \
+    types/float.c \
+    types/bigint.c \
+    types/blob.c \
+    types/decimal.c \
+    types/inet.c \
+    types/uuid_interface.c \
+    types/uuid.c \
+    types/timestamp.c \
+    types/timeuuid.c \
+    types/varint.c \
+    types/set.c \
+    types/map.c \
+    types/collection.c \
+  ";
+
+  case $(uname -s) in
+    Linux)
+      CASSANDRA_CFLAGS="-Wall -pedantic-errors -Wextra -Wno-long-long -Wno-deprecated-declarations -Wno-unused-parameter -Wno-variadic-macros -pthread"
+      ;;
+    Darwin)
+      CASSANDRA_CFLAGS="-Wall -pedantic -Wextra -Wno-long-long -Wno-deprecated-declarations -Wno-unused-parameter -Wno-variadic-macros"
+      ;;
+  esac
+  CASSANDRA_LIBS="-lssl -lz -luv -lstdc++"
```

Comment:
Right, I agree that we need to add checks for libuv since it is used in the driver, but we don't use any of the stdc++ things directly in the driver, that's why those flags were supplied from the outside for static linking the cpp-driver only